libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -266,6 +303,44 @@ std::string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const*
 	});
 }
 
+std::string YulUtilFunctions::requireWithErrorFunction(FunctionCall const& errorConstructorCall)
+{
+	ErrorDefinition const* errorDefinition = dynamic_cast<ErrorDefinition const*>(ASTNode::referencedDeclaration(errorConstructorCall.expression()));
+	solAssert(errorDefinition);
+
+	std::string const errorSignature = errorDefinition->functionType(true)->externalSignature();
+	// Note that in most cases we'll always generate one function per error definition,
+	// because types in the constructor call will match the ones in the definition. The only
+	// exception are calls with types, where each instance has its own type (e.g. literals).
+	std::string functionName = "require_helper_t_error_" + std::to_string(errorDefinition->id()) + "_" + errorDefinition->name();
+	for (ASTPointer<Expression const> const& argument: errorConstructorCall.sortedArguments())
+	{
+		solAssert(argument->annotation().type);
+		functionName += ("_" + argument->annotation().type->identifier());
+	}
+
+	std::vector<std::string> errorArgumentVars;
+	for (ASTPointer<Expression const> const& arg: errorConstructorCall.sortedArguments())
+	{
+		solAssert(arg->annotation().type);
+		if (arg->annotation().type->sizeOnStack() > 0)
+			errorArgumentVars += IRVariable(*arg).stackSlots();
+	}
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(condition <errorArguments>) {
+				if iszero(condition)
+					<revertWithError>
+			}
+		)")
+		("functionName", functionName)
+		("errorArguments", joinHumanReadablePrefixed(errorArgumentVars))
```

Comment:
This is what I meant in https://github.com/ethereum/solidity/pull/14913/files#r1576445073:

```suggestion
		// We're creating parameter names from the expressions passed into the constructor call,
		// which will result in odd names like `expr_29` that would normally be used for locals.
		// Note that this is the naming expected by `revertWithError()`.
		("functionParameterNames", joinHumanReadablePrefixed(errorArgumentVars))
```

Reply:
You missed the parameter name change: `errorArguments` -> `functionParameterNames`.

I guess I should have made it two separate comments, but thought it would be easier if I do it in one :(